### ISSUE TITLE
Not Working with Grafana v9.1.7 and GLPI 10.0.3

### ISSUE THREAD

**Issue by rodrigopadilh4** [ddurieux/glpi_app_grafana#81]
Hi,

After updating glpi to version 10 the plugin is not working. The data source is working, but the data is not displayed in the dashboard. 
Does this plugin work with this version?
glpi: 10.0.3
plugin glpi: 1.3.1
grafana: 9.1.7

Below image of the debugger

![image](https://user-images.githubusercontent.com/115184119/194400626-46262f3e-b785-426c-854a-dceb76dc06d2.png)

![image](https://user-images.githubusercontent.com/115184119/194401124-f907697a-3c42-41df-98d9-fadbf9ba1089.png)

![image](https://user-images.githubusercontent.com/115184119/194401392-ae657575-6b2c-43f0-b76b-e818436a9e0a.png)

**Comment by ddurieux** [comment]
Released for Grafana 9 : https://github.com/ddurieux/glpi_app_grafana/releases/download/2.1.0/ddurieux-glpi-app-2.1.0.zip

**Comment by rodrigopadilh4** [comment]
Thanks for the answer.

I installed the plugin by unzipping the zip file. The plugin shows that it is installed but does not show the datasource. I verified that the directory /grafana/plugins/ddurieux-glpi-app/datasource/ apparently lacks files comparing to version 1.3.1.

![image](https://user-images.githubusercontent.com/115184119/197894546-b988f181-45f9-4c55-b169-0c63093b64f4.png)

![image](https://user-images.githubusercontent.com/115184119/197894616-66acf039-612b-42c8-8c08-c798ad4c7d52.png)

![image](https://user-images.githubusercontent.com/115184119/197894182-927807b4-09ae-413f-84d7-a398bb43e97d.png)

**Comment by ddurieux** [comment]
You can't click on the glpi datasource?

**Comment by rodrigopadilh4** [comment]
No, he doesn't. In version 1.3 the datasource is automatically presented for configuration.

![image](https://user-images.githubusercontent.com/115184119/198043817-6763263f-fbd6-4aa8-805d-6d30b21c61b9.png)

**Comment by abreuti** [comment]
Yes, I have the same problem, my configuration show version 1.3.1

**Comment by edsontadeuas** [comment]
I´m experiencing the same problem too.

Grafana 9.2.4 + GLPI 10.0.5 is not working with glpi-plugin version 1.3.1.

So, when i did the upgrade to glpi-plugin version 2.1.0, the glpi datasource it not presented for configuration.

**Comment by DevilishDante** [comment]
Hello, I have the same problem too,
How to fix that ?
Thanks you to read me ;)

**Comment by ddurieux** [comment]
We made fixes for version 2.1.0 of the plugin, compatible with 9.5.x
Perhaps it could works too with v10.0
More news this week or next week

**Comment by edsontadeuas** [comment]
Hello ddurieux.

I just tested the latest plugin update (2.1.0) and it still doesn't work, grafana 9.2.4 + glpi 10.0.5.

Thanks!


![bug1](https://user-images.githubusercontent.com/41075289/203028738-ecf6f250-15cb-4c01-8922-878f6111468c.png)
![bug2](https://user-images.githubusercontent.com/41075289/203028740-c7e85916-a42f-4a3f-81a2-bc1ac684b13d.png)
![bug3](https://user-images.githubusercontent.com/41075289/203053857-17b66988-57e7-4f4c-9c31-c9b294e3fb91.png)

**Comment by ddurieux** [comment]
The version I re-uploaded this morning (2 hours ago)?

**Comment by edsontadeuas** [comment]
> The version I re-uploaded this morning (2 hours ago)?

Yes!

**Comment by ddurieux** [comment]
I think the URL in the panel is not right, can you check it?

**Comment by edsontadeuas** [comment]
> I think the URL in the panel is not right, can you check it?

This is the URL i´m using:

https://glpi.MYDOMAIN/front/ticket.php?is_deleted=0&as_map=0&criteria[0][field]=12&criteria[0][searchtype]=equals&criteria[0][value]=notold&search=search&itemtype=ticket&start=0



![bug4](https://user-images.githubusercontent.com/41075289/203054989-4d68aea9-cd19-4c61-ba26-a7c15bf73f73.png)

**Comment by ddurieux** [comment]
The URL is not right, you must have `itemtype=Ticket` and not `itemtype=ticket`, it's a huge difference ;)

**Comment by edsontadeuas** [comment]
> The URL is not right, you must have `itemtype=Ticket` and not `itemtype=ticket`, it's a huge difference ;)

I've tested both. The result is the same.

If i connect to my old GLPI (9.2.3), everything works ok.

https://glpi.MYDOMAIN/front/ticket.php?is_deleted=0&as_map=0&browse=0&criteria[0][link]=AND&criteria[0][field]=12&criteria[0][searchtype]=equals&criteria[0][value]=notold&itemtype=Ticket&start=0

**Comment by nrwillian** [comment]
@edsontadeuas 
You were able to install it from the zip provided by @ddurieux , I opened a question but for me the plugin has an invalid signature.

Already put the same as allow in grafana.ini but it still didn't work

**Comment by edsontadeuas** [comment]
> @edsontadeuas You were able to install it from the zip provided by @ddurieux , I opened a question but for me the plugin has an invalid signature.
> 
> Already put the same as allow in grafana.ini but it still didn't work

Yes, 

I tried to install the plugin available in grafana first, and then I downloaded the latest zip version available.
In both versions I was not successful in querying the GLPI.

There seems to be some incompatibility with the new API of GLPI 10 because it worked with GLPI 9.x.

**Comment by nrwillian** [comment]
> > @edsontadeuasVocê conseguiu instalá-lo a partir do zip fornecido por@ddurieux, abri uma pergunta, mas para mim o plugin está com uma assinatura inválida.
> > Já coloquei o mesmo como allow no grafana.ini mas mesmo assim não funcionou
> 
> Sim,
> 
> Tentei instalar o plugin disponível no grafana primeiro e depois baixei a versão zip mais recente disponível. Em ambas as versões não obtive sucesso na consulta ao GLPI.
> 
> Parece haver alguma incompatibilidade com a nova API do GLPI 10 porque funcionava com o GLPI 9.x.

Entendi, eu estou com problema na instalação. Para mim aparece plugin com assinatura inválida.

**Comment by edsontadeuas** [comment]
Let's wait for @ddurieux to evaluate this issue. 
Maybe he will release a version already compatible with glpi 10 soon.

*
Eu tive esse problema de assinatura inválida também, mas com a última versão que estava disponibilizada parou de acontecer.
Tenta baixar a versão direto do github e utiliza ela pra ver se você consegue ativar o plugin.

**Comment by nrwillian** [comment]
> Vamos esperar por@ddurieuxpara avaliar esta questão. Talvez ele lance uma versão já compatível com glpi 10 em breve.
> 
> 
> Eu tive esse problema de assinatura inválida também, mas com a última versão que estava disponibilizada parou de acontecer. Tente baixar a versão direto do github e utilize ela pra ver se você consegue ativar o plugin.

@edsontadeuas 

Fiz o download da versão do github, fiz o clone do repositório. E mesmo assim não foi. 

Tentei também com esse link https://github.com/ddurieux/glpi_app_grafana/releases/download/2.1.0/ddurieux-glpi-app-2.1.0.zip que o @ddurieux  disponibilizou nos comentários acima.

**Comment by ddurieux** [comment]
Hi all, I'm working on couple modifications / fixes, requested by Grafana team to be integrated into the catalog.
I will push a new version 2.1.0 in the week ;)

**Comment by DiegoRiopae** [comment]
@ddurieux também estou passando pelo mesmo problema que os colegas relatam. Estou usando a versão 2.10 que foi disponibilizada 

![erroGLPI](https://user-images.githubusercontent.com/119450285/204635052-2fbbc244-1d58-42c5-8fdd-0da50fb30c96.jpg)

**Comment by auguzsto** [comment]
First of all, thank you very much for your thoughtful work. I face the same problem here.

![image](https://user-images.githubusercontent.com/40308971/206757026-23c53d28-2e77-4f05-ae23-12ca21de53bf.png)

**Comment by abreuti** [comment]
Do someone already solved it? Can you help me?

**Comment by ddurieux** [comment]
The plugin os for the last stable version, GLPI 9.5 and GSIT 9.5

**Comment by c411n** [comment]
Hi @ddurieux 

I have the same problem with GLPI 10.0.6 and the latest version (2.0.1) of you plugin. I understand that it is a compatibility issue as it only works with glpi 9.5.x. 

Are there any plans to add compatibility with version 10.x.x ?

Thanks for advanced.

**Comment by miguelsilv6** [comment]
Any news for GLPI 10.x ? 

@ddurieux 

Thanks

**Comment by evandrofonseca22** [comment]
**Any updates? @ddurieux** 

I install the v2.1.0 and Grafana 10.0

My gpli uri: /front/ticket.php?is_deleted=0&as_map=0&browse=0&criteria[0][link]=AND&criteria[0][field]=1&criteria[0][searchtype]=contains&criteria[0][value]=REP&itemtype=Ticket

![image](https://github.com/ddurieux/glpi_app_grafana/assets/156134997/29413ae7-0662-4611-a6fa-ea746962c4f9)

**Comment by urbantao** [comment]
Hi, 
I'm trying to get some glpi 10.xxx stats into my grafana 10.0.2, I tried the plugin glpi 2.10 on grafana, but I only got some 400 errors. 
Here is my request : 
https://myglpi.dom.xxx/front/ticket.php?is_deleted=0&as_map=0&browse=0&criteria%5B0%5D%5Blink%5D=AND&criteria%5B0%5D%5Bfield%5D=12&criteria%5B0%5D%5Bsearchtype%5D=equals&criteria%5B0%5D%5Bvalue%5D=notold&itemtype=Ticket&start=0&_glpi_csrf_token=bcde4549df5f40b0a47f3d3f1efc0e264643f5e1f0ed3b566286df50d9df71ec&sort%5B%5D=19&order%5B%5D=DESC

is there something I can do, or maybe I am in the same situation than others, and the grafana glpi plugin does not work with glpi 10.xxx ?

**Comment by ddurieux** [comment]
Hi, no, all my developpements (grafana, plugins...) are stopped for GLPI 10, but will continue with GSIT only
